### DISCUSSION TITLE
x-data does not get updated in same element when externalized

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4389] [7. Bugs (archived)]
When I define `x-data` _inside_ the element itself, everything works fine, even when `x-init` is declared outside the element:

![image](https://github.com/user-attachments/assets/78b45ae6-aa90-41d8-95a1-7c20ec84a811)
![image](https://github.com/user-attachments/assets/9c90c873-b2b8-4380-9993-3a5e53e9e70f)

But as soon as I define `x-data` outside as well, it doesn't update the class in the element itself anymore, it still works on the child elements though at least where `active` changes to `true` after `list()` has been called.

![image](https://github.com/user-attachments/assets/e080c88c-1057-46ea-81a1-e60e89375141)

Why is that? Looks like a bug?

**Comment by trych** [comment]
Not sure if that answers your question, but I use this pattern, that I saw in the Alpine.bind() docs:

```js
Alpine.bind('SomeButton', () => ({
    'x-data'() { return {

            init() {
                // some init code, no need for an extra x-init prop
            },

            someOtherMethod() {
            
            }

        }
    },

    '@click'() {
        this.doSomething()
    },

    ':disabled'() {
        return this.shouldDisable
    },
}))
```

**Reply by simonferndriger** [reply]
I like the syntax. However, it doesn't solve the underlying issue unfortunately...

**Comment by ekwoka** [comment]
This may be a bug with how init is handled inside x-bind...
